lib/workers/single.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const bluebird = require('bluebird');
+const Popper = require('./popper.js').Popper;
+
+function Worker(client, config) {
+  this.client = client;
+  let queues = config.queues;
+  let interval = config.interval || 60000;
+  this.popper = new Popper(client, queues, interval);
+}
+
+/**
+ * Stop the worker.
+ *
+ * This does not preempt any work -- it just causes the `run` promise to resolve
+ * after the current job (if any) is finished.
+ */
+Worker.prototype.stop = function() {
+  this.popper.stop();
+}
+
+/**
+ * Process jobs.
+ *
+ * Returns a promise that resolves when the worker is stopped. Throws when there
+ * is an unrecoverable exception.
+ */
+Worker.prototype.run = function() {
+  let self = this;
+  return bluebird.Promise.try(() => {
+    return self.popper.pop()
+      .then((job) => {
+        if (job) {
+          return job.process()
+            .then(() => {
+              return self.run();
+            });
+        } else {
+          return;
+        }
+      })
+  })
```

Comment:
The linting wants all branches of functions to return, so it'd have to be a ternary operator, which starts to get a little long. Even so, I did incorporate your suggestion about the invocation to `this.run`